src/Core/src/Graphics/PaintExtensions.cs

```diff
@@ -25,18 +25,18 @@ public static partial class PaintExtensions
 		public static bool IsNullOrEmpty([NotNullWhen(true)] this Paint? paint)
 		{
 			if (paint is SolidPaint solidPaint)
-				return solidPaint == null || solidPaint.Color == null;
+				return solidPaint is null || solidPaint.Color is null;
 
 			if (paint is GradientPaint gradientPaint)
-				return gradientPaint == null || gradientPaint.GradientStops.Length == 0;
+				return gradientPaint is null || gradientPaint.GradientStops.Length == 0 || (gradientPaint.StartColor is null && gradientPaint.EndColor is null);
```

Comment:
@jsuarezruiz this additional bracket does not solve the issue when only one color does not exist/work